mlir/include/mlir/Dialect/Vector/Utils/VectorUtils.h

```diff
@@ -211,6 +211,11 @@ Value createReadOrMaskedRead(OpBuilder &builder, Location loc, Value source,
 ///      static sizes in `shape`.
 LogicalResult isValidMaskedInputVector(ArrayRef<int64_t> shape,
                                        ArrayRef<int64_t> inputVectorSizes);
+
+/// Returns true if the leading dim(s) of `type` are fixed and the trailing dim
+/// is scalable.
+bool isTrailingDimScalable(VectorType type);
```

Comment:
I think it'd be a little clearer to add 'only':
```suggestion
bool isOnlyTrailingDimScalable(VectorType type);
```